evmcore/state_processor_test.go

```diff
@@ -298,6 +301,101 @@ func TestProcess_TracksParentBlockHashIfPragueIsEnabled(t *testing.T) {
 	}
 }
 
+func TestRlpEncodedMaxHeaderSizeInBytes_IsAnUpperBound(t *testing.T) {
+	setToMax := func(b []byte) []byte {
+		for i := range b {
+			b[i] = 0xFF
+		}
+		return b
+	}
+
+	maxHash := common.Hash(setToMax(make([]byte, 32)))
+	maxAddress := common.Address(setToMax(make([]byte, 20)))
+	maxBloom := types.Bloom(setToMax(make([]byte, 2048)))
+	maxBlock := types.BlockNonce(setToMax(make([]byte, 8)))
+	maxUint64 := uint64(math.MaxUint64)
+
+	// Geth sanity check for extra defines an upper bound of 100 * 1024 bytes.
+	// Sonic uses the extra field to save time and duration.
+	extra := inter.EncodeExtraData(
+		time.Unix(math.MaxInt64, math.MaxInt64),
+		time.Duration(math.MaxInt64)*time.Nanosecond,
+	)
+
+	header := &types.Header{
+		ParentHash:       maxHash,
+		UncleHash:        maxHash,
+		Coinbase:         maxAddress,
+		Root:             maxHash,
+		TxHash:           maxHash,
+		ReceiptHash:      maxHash,
+		Bloom:            maxBloom,
+		Difficulty:       big.NewInt(math.MaxInt64),
+		Number:           big.NewInt(math.MaxInt64),
+		GasLimit:         math.MaxUint64,
+		GasUsed:          math.MaxUint64,
+		Time:             math.MaxUint64,
+		Extra:            extra,
+		MixDigest:        maxHash,
+		Nonce:            maxBlock,
+		BaseFee:          big.NewInt(math.MaxInt64),
+		WithdrawalsHash:  &maxHash,
+		BlobGasUsed:      &maxUint64,
+		ExcessBlobGas:    &maxUint64,
+		ParentBeaconRoot: &maxHash,
+		RequestsHash:     &maxHash,
+	}
+
+	data, err := rlp.EncodeToBytes(header)
+	require.NoError(t, err)
+	require.LessOrEqual(t, uint64(len(data)), RlpEncodedMaxHeaderSizeInBytes, "header exceeds maximum size")
+}
+
+func TestProcess_BrioLimitsBlockSize(t *testing.T) {
+	for _, isBrio := range []bool{false, true} {
+		t.Run(fmt.Sprintf("isBrio=%v", isBrio), func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			runner := NewMock_transactionRunner(ctrl)
+
+			transactions := []*types.Transaction{
+				types.NewTx(&types.LegacyTx{Data: make([]byte, params.MaxBlockSize/2)}),
+				types.NewTx(&types.LegacyTx{Data: make([]byte, params.MaxBlockSize/2)}),
+			}
+
```

Comment:
could you add her e something like `require.greater(tx1.size+tx2.size, params.MaxBlockSize)`?